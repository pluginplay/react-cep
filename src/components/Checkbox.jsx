import styled from 'styled-components'
import React from 'react'
import omit from 'lodash/omit'

class Checkbox extends React.Component {
  render () {
    return (
      <div className={this.props.className}>
        <input type={'checkbox'} {...omit(this.props, ['className'])} />
        <span />
      </div>
    )
  }
}

export default styled(Checkbox)`
  width: 16px;
  height: 16px;
  position: relative;
  outline: none;
  input[type=checkbox] {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    opacity: 0;
  }
  span {
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${props => props.theme.background.dark};
    border: solid 1px ${props => props.theme.background.light};
    transition: border-color 0.1s ease-in-out;
    &:before {
      display: block;
      content: ' ';
      position: absolute;
      top: 4px;
      left: 4px;
      right: 4px;
      bottom: 4px;
      transform: scale(0);
      transition: transform 0.1s ease-in-out;
      background: ${props => props.theme.ae.systemHighlightColor};
    }
  }
  &:hover span, input[type=checkbox]:focus + span {
    border-color: ${props => props.theme.background.lighter};
  }
  input[type=checkbox]:checked + span:before { transform: scale(1); }
  input[type=checkbox]:focus + span:before, &:hover span:before { transform: scale(0.5); }
  &:hover input[type=checkbox]:checked + span:before, input[type=checkbox]:focus:checked + span:before { transform: scale(1.2); }
`
