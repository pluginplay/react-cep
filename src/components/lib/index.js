import SystemState from '../../models/SystemState'
import PersistedState from '../../models/PersistedState'

export { SystemState, PersistedState }
