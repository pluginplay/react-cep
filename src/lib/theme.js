import * as csInterface from 'cep-lib/csinterface'
import { lighten, darken } from 'polished'

export const getDefaultTheme = () => {
  const ae = csInterface.getThemeInformation()
  ae.systemHighlightColor = '#0078d7' // We have to hard-code this because Windows gets the wrong color.
  return {
    ae,
    background: {
      lighter: lighten(0.2, ae.panelBackgroundColor),
      light: lighten(0.1, ae.panelBackgroundColor),
      dark: darken(0.05, ae.panelBackgroundColor),
      darker: darken(0.1, ae.panelBackgroundColor)
    },
    colors: {
      green: '#23EB87',
      red: '#eb093f',
      blue: ae.systemHighlightColor
    }
  }
}
