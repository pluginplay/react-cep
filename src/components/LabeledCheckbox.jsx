import React from 'react'
import Checkbox from './Checkbox'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import omit from 'lodash/omit'

class LabeledCheckbox extends React.Component {
  render () {
    return (
      <label className={this.props.className}>
        <Checkbox {...omit(this.props, ['className'])} />
        <span>{this.props.label}</span>
      </label>
    )
  }
}

LabeledCheckbox.propTypes = {
  label: PropTypes.string
}

export default styled(LabeledCheckbox)`
  display: block;
  margin: 10px 0;
  &:first-child { margin-top: 0; }
  > span, div {
    display: inline-block;
    vertical-align: middle;
  }
  > span {
    margin-left: 7px;
  }
`
