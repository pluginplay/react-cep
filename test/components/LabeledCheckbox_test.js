import { withTheme, expect } from '../setup'
import React from 'react'
import LabeledCheckbox from '../../src/components/LabeledCheckbox'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { getDefaultTheme } from '../../src/lib/theme'

import Checkbox from '../../src/components/Checkbox'

it('renders correctly', () => {
  const tree = renderer
    .create(withTheme(<LabeledCheckbox checked={false} label={'Check me baby'} />))
    .toJSON()
  jestExpect(tree).toMatchSnapshot()
})

it('forwards props', () => {
  const wrapper = shallow(<LabeledCheckbox theme={getDefaultTheme()} foo={'bar'} />).dive()
  expect(wrapper.find(Checkbox).first()).to.have.prop('foo').equal('bar')
  expect(wrapper.find(Checkbox).first()).to.not.have.prop('className')
})
