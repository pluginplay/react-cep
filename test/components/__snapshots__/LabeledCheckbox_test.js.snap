// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className="sc-bwzfXH iPQAKy"
>
  <div
    className="sc-bdVaJa cNLavf"
  >
    <input
      checked={false}
      label="Check me baby"
      type="checkbox"
    />
    <span />
  </div>
  <span>
    Check me baby
  </span>
</label>
`;
